typescript/packages/plugins/ironclad/src/ironclad.service.ts

```diff
@@ -0,0 +1,735 @@
+                collateralBalanceResult as { value: bigint }
+            ).value;
+
+            if (collateralBalance > 0n) {
+                // Withdraw all collateral
+                await walletClient.sendTransaction({
+                    to: vaultAddress,
+                    abi: IC_VAULT_ABI,
+                    functionName: "withdraw",
+                    args: [collateralBalance],
+                });
+            }
+
+            return txHash.hash;
+        } catch (error) {
+            throw Error(`Failed to close Trove: ${error}`);
+        }
+    }
+
+    @Tool({
+        name: "monitor_position_ironclad",
+        description: "Monitor health of a Trove position",
+    })
+    async monitorPosition(
+        walletClient: EVMWalletClient,
+        parameters: MonitorPositionParameters
+    ): Promise<{
+        currentCollateral: string;
+        currentDebt: string;
+        troveStatus: string;
+    }> {
+        try {
+            const vaultAddress = getVaultAddress(parameters.tokenAddress);
+
+            // Get token decimals
+            const decimals = Number(
+                await walletClient.read({
+                    address: parameters.tokenAddress as `0x${string}`,
+                    abi: ERC20_ABI,
+                    functionName: "decimals",
+                })
+            );
+
+            // Get Trove status
+            const statusResult = await walletClient.read({
+                address: TROVE_MANAGER_ADDRESS as `0x${string}`,
+                abi: TROVE_MANAGER_ABI,
+                functionName: "getTroveStatus",
+                args: [walletClient.getAddress(), vaultAddress],
+            });
+            const status = Number((statusResult as { value: bigint }).value);
+
+            // Get Trove collateral
+            const collateralResult = await walletClient.read({
+                address: TROVE_MANAGER_ADDRESS as `0x${string}`,
+                abi: TROVE_MANAGER_ABI,
+                functionName: "getTroveColl",
+                args: [walletClient.getAddress(), vaultAddress],
+            });
+            const collateral = (collateralResult as { value: bigint }).value;
+
+            // Get Trove debt
+            const debtResult = await walletClient.read({
+                address: TROVE_MANAGER_ADDRESS as `0x${string}`,
+                abi: TROVE_MANAGER_ABI,
+                functionName: "getTroveDebt",
+                args: [walletClient.getAddress(), vaultAddress],
+            });
+            const debt = (debtResult as { value: bigint }).value;
+
+            // Map status number to string
+            const statusMap = {
+                0: "nonExistent",
+                1: "active",
+                2: "closedByOwner",
+                3: "closedByLiquidation",
+                4: "closedByRedemption",
+            };
+
+            return {
+                currentCollateral: collateral.toString(),
+                currentDebt: formatUnits(debt, 18),
```

Comment:
No need to format this. You can just say its in basis units and the LLM should figure out how to transform it into decimals